fe/fe-core/src/main/java/com/starrocks/http/rest/v2/TablePartitionAction.java

```diff
@@ -0,0 +1,93 @@
+import com.starrocks.privilege.AccessDeniedException;
+import io.netty.handler.codec.http.HttpMethod;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.BooleanUtils;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static com.starrocks.catalog.InternalCatalog.DEFAULT_INTERNAL_CATALOG_NAME;
+
+public class TablePartitionAction extends TableBaseAction {
+
+    private static final String TEMPORARY_KEY = "temporary";
+
+    public TablePartitionAction(ActionController controller) {
+        super(controller);
+    }
+
+    public static void registerAction(ActionController controller) throws IllegalArgException {
+        controller.registerHandler(
+                HttpMethod.GET,
+                String.format(
+                        "/api/v2/catalogs/{%s}/databases/{%s}/tables/{%s}/partition", CATALOG_KEY, DB_KEY, TABLE_KEY),
+                new TablePartitionAction(controller));
+    }
+
+    @Override
+    protected void doExecuteWithoutPassword(
+            BaseRequest request, BaseResponse response) throws AccessDeniedException {
+        String catalogName = getSingleParameterOrDefault(
+                request, CATALOG_KEY, DEFAULT_INTERNAL_CATALOG_NAME, Function.identity());
+        String dbName = getSingleParameterRequired(request, DB_KEY, Function.identity());
+        String tableName = getSingleParameterRequired(request, TABLE_KEY, Function.identity());
+        boolean temporary = getSingleParameterOrDefault(request, TEMPORARY_KEY, false, BooleanUtils::toBoolean);
+        int pageNum = getPageNum(request);
+        int pageSize = getPageSize(request);
+
+        PagedResult<TablePartitionView> pagedResult = new PagedResult<>();
+        pagedResult.setPageNum(pageNum);
+        pagedResult.setPageSize(pageSize);
+        pagedResult.setPages(0);
+        pagedResult.setTotal(0);
+        pagedResult.setItems(new ArrayList<>(0));
+
+        OlapTable olapTable = getOlapTable(catalogName, dbName, tableName);
+        Collection<Partition> partitions =
+                temporary ? olapTable.getTempPartitions() : olapTable.getPartitions();
+        if (CollectionUtils.isNotEmpty(partitions)) {
+            int pages = partitions.size() / pageSize;
+            pagedResult.setPages(partitions.size() % pageSize == 0 ? pages : (pages + 1));
+            pagedResult.setTotal(partitions.size());
+            pagedResult.setItems(partitions.stream()
+                    .sorted(Comparator.comparingLong(Partition::getId))
+                    .skip((long) pageNum * pageSize)
+                    .limit(pageSize)
+                    .map(partition ->
+                            TablePartitionView.createFrom(olapTable.getPartitionInfo(), partition))
+                    .collect(Collectors.toList())
+            );
+        }
```

Comment:
need read lock

Reply:
Yeah, It is a problem, and i encapsulated the processing logic of OlapTable into the db read lock.